elements/lisk-transactions/test/13_vote_transaction.spec.ts

```diff
@@ -0,0 +1,158 @@
+			);
+			it.todo('should create vote object when it does not exist before');
+			it.todo('should update vote object when it exists before');
+		});
+
+		describe('when asset.votes contain negative amount which makes account.votes to be 0 entries', () => {
+			it.todo('should not return error');
+			it.todo('should not change account balance');
+			it.todo('should remove vote which has zero amount');
+			it.todo('should update vote which has non-zero amount');
+			it.todo('should make account to have correct unlocking');
+			it.todo('should order account.unlocking');
+			it.todo(
+				'should make downvoted delegate account to have correct totalVotesReceived',
+			);
+		});
+
+		describe('when asset.votes contain negative and positive amount which makes account.votes to be 10 entries', () => {
+			it.todo('should not return error');
+			it.todo('should make account to have correct balance');
+			it.todo('should make account to have correct unlocking');
+			it.todo('should order account.votes');
+			it.todo('should order account.unlocking');
+			it.todo(
+				'should make upvoted delegate account to have correct totalVotesReceived',
+			);
+			it.todo(
+				'should make downvoted delegate account to have correct totalVotesReceived',
+			);
+		});
+
+		describe('when asset.votes contain delegate address which is not registered', () => {
+			it.todo('should return errors');
+		});
+
+		describe('when the last asset.votes amount makes sender not having sufficient balance', () => {
+			it.todo('should return errors');
+		});
+
+		describe('when asset.votes positive amount makese account.votes entries more than 10', () => {
+			it.todo('should return errors');
+		});
+
+		describe('when asset.votes negative amount decrease acount.votes entries yet positive amount makes account exceeds more than 10', () => {
+			it.todo('should return errors');
+		});
+
+		describe('when asset.votes negative amount and makes account.unlocking more than 20 entries', () => {
+			it.todo('should return errors');
+		});
+
+		describe('when asset.votes negative amount exceeds the previously voted amount', () => {
+			it.todo('should return errors');
+		});
+	});
+
+	describe('undoAsset', () => {
+		describe('when asset.votes contain positive amount which makes account.votes to be 10 entries', () => {
+			it.todo('should not return error');
+			it.todo('should make account to have original values before apply');
+			it.todo(
+				'should make upvoted delegate account to have original values before apply',
+			);
+			it.todo(
+				'should make downvoted delegate account to have original values before apply',
+			);
+		});
+
+		describe('when asset.votes contain negative amount which makes account.votes to be 0 entries', () => {
+			it.todo('should not return error');
+			it.todo('should make account to have original values before apply');
+			it.todo(
+				'should make downvoted delegate account to have original values before apply',
+			);
```

Comment:
when undoing a negative vote, we also need to remove the "unlocking object" that was created. (maybe this is understood as "make account have orginal values", in which case, disregard this comment :) )

Reply:
yes it will be checked with the account exact equal, so this case should be covered